ios/Framework/Core/Base/BaseScreenlet.swift

```diff
@@ -328,24 +328,26 @@ import QuartzCore
 	//MARK: HUD methods
 
 	public func showHUDWithMessage(message: String?,
-			closeMode: ProgressCloseMode,
-			spinnerMode: ProgressSpinnerMode) {
-
-		_progressPresenter?.showHUDInView(rootView(self),
-			message: message,
-			closeMode: closeMode,
-			spinnerMode: spinnerMode)
-	}
-
-	public func showHUDAlert(message message: String) {
-		_progressPresenter?.showHUDInView(rootView(self),
-			message: message,
-			closeMode: .ManualClose_TouchClosable,
-			spinnerMode: .NoSpinner)
+			forInteractor interactor: Interactor) {
+        
+        _progressPresenter?.showHUDInView(rootView(self),
+            message: message,
+            forInteractor: interactor)
 	}
 
-	public func hideHUD() {
-		_progressPresenter?.hideHUD()
+    public func hideHUDWithMessage(message: String?,
+            forInteractor interactor: Interactor,
+            withError error: NSError?) {
+        var view: UIView?
+        
+        if let message = message {
+            view = rootView(self)
```

Comment:
That is because the original hideHUD (the one that only hide the progress) didn't need the view. This method (called from hideInteractorHUD) will show another progress only if there is a message (not nil), otherwise it will only hide the view presented in showHUDWithMessage, so it doesn't need the UIView.